lib/pacemaker/pcmk_simulate.c

```diff
@@ -362,15 +362,20 @@ profile_file(const char *xml_file, long long repeat,
     }
 
     for (int i = 0; i < repeat; ++i) {
+        int rc;
         xmlNode *input = cib_object;
 
         if (repeat > 1) {
             input = pcmk__xml_copy(NULL, cib_object);
         }
         scheduler->input = input;
         set_effective_date(scheduler, false, use_date);
-        pcmk__schedule_actions(input, scheduler_flags, scheduler);
+        rc = pcmk__schedule_actions(input, scheduler_flags, scheduler);
         pe_reset_working_set(scheduler);
+
+        if (rc != pcmk_rc_ok) {
+            break;
+        }
```

Comment:
Since this is for profiling, I think we can ignore the return code (if for some reason we're testing bad input, we'd want to test it the same number of times as everything else)